blog/en/blog/2022/09/23/build-event-driven-api.md

```diff
@@ -0,0 +1,126 @@
+
+Although the three concepts are independent, they complement each other well. This blog post explores how to build event-driven API Services using these **3 well-known patterns** to build a highly scalable and [distributed system](https://en.wikipedia.org/wiki/Distributed_computing). We will break down each concept and understand the role of each in our particular approach.
+
+## The event-driven architecture
+
+An **event-driven architecture** makes use of events to trigger and communicate between decoupled services.  Each service publishes an event whenever it updates its data. Other services subscribe to events. When an event is received, a service updates its data.
+
+This architecture has several benefits such as you completely **decoupling** producer and consumer services.  If one service has a failure, the rest will keep running. Consumers can respond to events immediately as they arrive. It adds **agility** as well. If you want to add another service, you can just have it subscribe to an event and have it generate new events of its own. The existing services don’t know or care that this has happened, so there’s no impact on them.
+
+![Event driven diagram](https://static.apiseven.com/2022/09/20/63289f6bb3090.png)
+
+## Why not use simply CRUD❓😕
+
+Usually, we use the same data model to query and update a database that is similar to the basic **CRUD** operations (**“CREATE”, “READ”, “UPDATE”, and “DELETE”**) and it is the most straightforward way of dealing with data manipulation. We can build API services by following this simple principle. Any tool or framework that advertises itself as a quick method to bring your application to the market. But modern applications involve more complex business processes with workflows, validation, and business logic that are difficult to express using the classic CRUD paradigm.
+
+Some of the following challenges you can think of:
+
+-⛔️ Since for both read and write operations, and the same DTO or data transfer object are used, there’s a chance that read and write operations will be out of sync.
```

Comment:
```suggestion
- ⛔️ Since for both read and write operations, and the same DTO or data transfer object are used, there’s a chance that read and write operations will be out of sync.
```

Reply:
Add a space, otherwise the Markdown format will disapper
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/39793568/192211737-38988ccb-970e-46fb-8653-723b36a1b441.png">